Python/Product/TestAdapter/testlauncher.py

```diff
@@ -73,12 +74,21 @@ def load_debugger(secret, port, debugger_search_path):
             input()
         sys.exit(1)
 
-def run(testRunner, args):
+def run(testRunner,test_list, args):
     """Runs the test
     testRunner -- test runner to be used `pytest` or `nose`
     args -- arguments passed into the test runner
     """
 
+    all_tests = []
+    if test_list and os.path.exists(test_list):
+        with open(test_list, 'r', encoding='utf-8') as tests:
+            all_tests.extend(t.strip() for t in tests)
```

Comment:
I think i'll change this to append to args directly.. ie. remove all_test variable